guidelines/coding-guidelines.md

```diff
@@ -11,3 +11,59 @@ For public facing interfaces we favor read-only collections and enumerables. For
 * Avoid using `System.Linq`
 * Avoid using `foreach` over collections that do not have a struct enumerator
 
+## Dependencies
+
+### 3rd party dependencies
+
+#### Package dependencies
+
+We want to compile against the lowest major to catch breaking changes
+
+PackageReference: `[Major.0.0, Major+1)`
+
+#### Tests
+
+We want to compile against the latest version to catch breaking changes and bugs. This is the likely version that our users would be on.
+
+PackageReference: `[Major.*, Major+1)`
+
+### Upstream core dependency
+
+#### Stable release of core
+
+We want to compile against the lowest major to catch breaking changes
+
+PackageReference: `[Major.0.0, Major+1)`
+
+#### During release preparation of core (most likely majors)
+
+We want downstreams to pick up the latest core alpha/beta/rc immediately to quickly catch problems.
+
+PackageReference: `[Major.0.0-*, Major+1)`
```

Comment:
In RabbitMQ we compile against a specific alpha, but after more thought, I'm on the fence about this. If the general vote is for a wildcard here I'm not opposed.

@Particular/rabbitmq-transport-maintainers thoughts?